osu.Game/Utils/BindableValueAccessor.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.TypeExtensions;
+
+namespace osu.Game.Utils
+{
+    internal static class BindableValueAccessor
+    {
+        private static readonly MethodInfo get_method = typeof(BindableValueAccessor).GetMethod(nameof(getValue), BindingFlags.Static | BindingFlags.NonPublic)!;
+        private static readonly MethodInfo set_method = typeof(BindableValueAccessor).GetMethod(nameof(setValue), BindingFlags.Static | BindingFlags.NonPublic)!;
+
+        public static object GetValue(IBindable bindable)
+        {
+            Type? bindableWithValueType = bindable.GetType().GetInterfaces().FirstOrDefault(isBindableT);
+            if (bindableWithValueType == null)
+                return bindable;
+
+            return get_method.MakeGenericMethod(bindableWithValueType.GenericTypeArguments[0]).Invoke(null, [bindable])!;
+        }
+
+        public static void SetValue(IBindable bindable, object value)
+        {
+            Type? bindableWithValueType = bindable.GetType().EnumerateBaseTypes().FirstOrDefault(isBindableT);
+            if (bindableWithValueType == null)
+                return;
+
+            set_method.MakeGenericMethod(bindableWithValueType.GenericTypeArguments[0]).Invoke(null, [bindable, value]);
+        }
+
+        private static bool isBindableT(Type type)
+            => type.IsGenericType
+               && (type.GetGenericTypeDefinition() == typeof(Bindable<>)
+                   || type.GetGenericTypeDefinition() == typeof(IBindable<>));
```

Comment:
For anyone else that might be as confused as me, no this is not the same thing twice, note the `Bindable<>` vs `IBindable<>` in there...

Reply:
I can inline it in each method if you'd prefer. I originally had it that way.

Reply:
Combined with the other comment, if you inlined just the relevant check (i.e. `IBindable<>` in `GetValue()` and `Bindable<>` in `SetValue()`), I think that could be a net readability gain, yeah - unless there's some associated reason why you didn't do that.

Reply:
Nope - just a case of trying to be too smart.